Locale related changes

jacqui opened the pull request:
This PR has several changes and is a bit lengthy, but the gist is that it takes us closer to supporting multilingual content editing and publishing for the tiny news sites, issues #118, #119 and #120.

The work in this PR takes us through the basics of supporting i18n and multilingual publishing. Details are listed below. I'd like to start up the next round of work in separate PRs (issues #126, #127, #128, #129).

**To test**: open script editor, test as add-on using latest code. Before selecting a locale, the locale picker should display in the Publishing Tools sidebar. After selecting a locale, the chosen locale should be displayed as text in the sidebar. Updating a document works as expected. Associating a new document in another locale preserves the contents of both or all locales.

* [x] adds locales list to `getArticleMeta` output
* [x] adds selected localeID to `getArticleMeta` output (will be NULL if none set)
* [x] adds a locale dropdown to the sidebar with available locales for the webiny install listed
  * [x] selects the article's localeID if it's set
  * [x] selects the default locale (as defined in webiny) otherwise
* [x] stores the selected locale as a document (google) property (`storeLocaleID`, function written ages ago)
* [x] retrieves latest article content before saving
* [x] if the **article** content is in multiple languages, preserves content in other locales
* [x] overwrites (updates) content in the same locale as the current document
* [x] preserves and properly updates content in the same manner for **pages**
* [x] dropdown only appears in sidebar if locale has not been selected
* [x] otherwise, after locale is selected, dropdown is replaced by locale code (ex: `en-US`) text
* [x] there is currently no way to change the locale of an article/document once the locale has been set 
  * we might want to offer this in the future but I'm not sure what the use case would be
* [x] backend code for the sidebar no longer sets the locale to the default by, um, default
  * this indirectly happens via the sidebar, in that the dropdown is preselected to the default locale, and will thus be submitted as-is if you don't otherwise change it

TylerFisher commented:
Tried to test this but it mostly didn't work as described. Perhaps "latest code" isn't the latest? My chosen locale never becomes the "selected" locale, I always have the option to switch. A lot of data never gets submitted, see #130.

jacqui commented:
Ack, sorry about that! I think I tried to do too much in this PR, got confused along the way, and then didn't test it all, on a fresh document, at the end. 

I'm working through the issues and fixing them as I go. So far:

* [x] headline wasn't being set on publish or preview calls - only when first naming the document
  * I changed this so every time a document is saved (that's preview or publish), the headline is updated with whatever is in the Headline form field.
  * I'm sure this was how I originally coded it, and I think it was possibly lost via code shuffling or something
* [x] locale ID wasn't being stored on the document properties
  * this I'm sure is because of me shuffling around code and trying to get my head around how to handle i18n across multiple google docs
  * I've added a call to `storeLocaleID()` on every save; that'll either come from the form dropdown, or it will be storing the same localeID as previously set
* [ ] **QUESTION** should the slug update when the headline changes? I have this intentionally coded to never change once set but I could switch how it works for sure
* [ ] should be able to select authors and have that selection preserved on the document
* [ ] should be able to select tags and have that selection preserved on the document

TylerFisher commented:
Good question. I think it should change until you click publish for the first time (so if you preview a few times and try different headlines, the slug changes too). Once you've published, you're probably linking to it somewhere, so we don't want that to break.

jacqui commented:
> Good question. I think it should change until you click publish for the first time (so if you preview a few times and try different headlines, the slug changes too). Once you've published, you're probably linking to it somewhere, so we don't want that to break.

Okay, makes sense. I'll get to this tomorrow.
